models/silence.go

```diff
@@ -51,7 +52,12 @@ func (s *Silence) Matches(alert string, tags opentsdb.TagSet) bool {
 }
 
 func (s Silence) ID() string {
+	if s.CachedId != "" {
+		return s.CachedId
+	}
 	h := sha1.New()
 	fmt.Fprintf(h, "%s|%s|%s%s", s.Start, s.End, s.Alert, s.Tags)
-	return fmt.Sprintf("%x", h.Sum(nil))
+	id := fmt.Sprintf("%x", h.Sum(nil))
```

Comment:
So we're still using the un-ordered hash map, though? Are we actually fixing this?

Reply:
Should still work. The ID generation should only happen for new silences; for existing ones, the `getSilences` function should fill out the ID with the real one in redis.